src/libs/banners/banners.ts

```diff
@@ -177,3 +178,44 @@ export const getNetworksWithFailedRPCBanners = ({
       }
     })
 }
+
+export const getNetworksWithCriticalPortfolioErrorBanners = ({
+  selectedAccount,
+  networks,
+  portfolio
+}: {
+  selectedAccount: AccountId | null
+  networks: NetworkDescriptor[]
+  portfolio: PortfolioController
+}): Banner[] => {
+  if (!selectedAccount) return []
+
+  const selectedAccPortfolio = portfolio.latest[selectedAccount]
+  if (!selectedAccPortfolio) return []
+
+  return Object.keys(selectedAccPortfolio)
+    .filter((networkId) => portfolio.networksWithAssets.includes(networkId))
+    .flatMap((network) => {
+      const portfolioForNetwork = selectedAccPortfolio[network]
+
+      const criticalError = portfolioForNetwork?.criticalError
+      if (!criticalError) return []
+
+      const networkData = networks.find((n: NetworkDescriptor) => n.id === network)
+      if (!networkData) return []
```

Comment:
shouldn't we return an issue in this case

Reply:
yes, good catch, error log added ✅